src/Microsoft.FeatureManagement.AspNetCore/TargetingHttpContextMiddleware.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+
+using Microsoft.AspNetCore.Http;
+using Microsoft.FeatureManagement.FeatureFilters;
+using System;
+using System.Threading.Tasks;
+
+namespace EvaluationDataToApplicationInsights.Telemetry
+{
+    /// <summary>
+    /// Used to add targeting information to http context. This allows synronous code to access targeting information.
+    /// </summary>
+    public class TargetingHttpContextMiddleware
+    {
+        private readonly RequestDelegate _next;
+
+        /// <summary>
+        /// Creates an instance of the TargetingHttpContextMiddleware
+        /// </summary>
+        public TargetingHttpContextMiddleware(RequestDelegate next) { 
+            _next = next; 
+        }
+
+        /// <summary>
+        /// Adds targeting information to the http context.
+        /// </summary>
+        /// <param name="context">The <see cref="HttpContext"/> to add the targeting information to.</param>
+        /// <param name="targetingContextAccessor">The <see cref="ITargetingContextAccessor"/> to retrieve the targeting information from.</param>
+        /// <exception cref="ArgumentNullException">Thrown if the provided context or targetingContextAccessor is null.</exception>
+        public async Task InvokeAsync(HttpContext context, ITargetingContextAccessor targetingContextAccessor)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            if (targetingContextAccessor == null)
+            {
+                throw new ArgumentNullException(nameof(targetingContextAccessor));
+            }
+
+            TargetingContext targetingContext = await targetingContextAccessor.GetContextAsync().ConfigureAwait(false);
+
+            if (targetingContext != null)
+            {
+                context.Items["TargetingId"] = targetingContext.UserId;
```

Comment:
It seems like the context.Items["TargetingId"] is expected to consume by the TargetingTelemetryInitializer. 
So maybe we should seperate the keyname to a constant string or even put it in a file e.g. TelemetryProperties.cs. "FeatureName", "Enabled", "TargetingId", "Variant" and "VariantAssignmentReason" should also be put in that file.

My another concern is that httpcontext is a shared place which can be modified by any custom middlewares. So the question is whether "TargetingId" is a concept which will only be used by FeatureManagement lib. What if users want to set their own "TargetingId" in HttpContext. This may not be a valid user case. But we have a precedent for feature filter settings cache which will be stored in the shared memory cache. What we did is to add a "FeatureManagement" prefix to the key name: https://github.com/microsoft/FeatureManagement-Dotnet/blob/preview/src/Microsoft.FeatureManagement/FeatureManager.cs#L666

Reply:
> It seems like the context.Items["TargetingId"] is expected to consume by the TargetingTelemetryInitializer.
So maybe we should seperate the keyname to a constant string or even put it in a file e.g. TelemetryProperties.cs. "FeatureName", "Enabled", "TargetingId", "Variant" and "VariantAssignmentReason" should also be put in that file.

This seemed like a good idea, so I started adding it. The issue is the publisher and initializer are in different projects. We could make TelemetryProperties but it would have it be public which is an interface change I doubt we want.

Reply:
> What we did is to add a "FeatureManagement" prefix to the key name

Went ahead and added `Microsoft.FeatureManagement.` as a prefix.